packages/block-library/src/query-pagination/style.scss

```diff
@@ -13,6 +13,14 @@ $pagination-margin: 0.5em;
 			margin-right: 0;
 		}
 	}
+
+	// This moves the next link to the right side of the container,
+	// which is important when it's the only block displayed
+	// and the block has a "space-between" justification.
+	&.is-content-justification-space-between > .wp-block-query-pagination-next {
```

Comment:
This cover a super specific use case and only using css. It feels better to have themes add styles like that for now, no?

Reply:
I think it's justified because the default behavior looks kind of bad and you are basically forcing every theme to add this CSS to have a decent default, rather than a customization.

Reply:
Still, I'm not sure about the approach. If themers  insist that this is an acceptable stop gap solution we should add a comment that this is a temp solution and link to the main issue. Also have you tested the above PR I linked? I should have linked that to the issue before... I had linked it with the alternative PRs, sorry about that..

Reply:
When would you ever want a pagination pointing to "next" to be on the left? 

I guess we should consider RTL though...

Reply:
Why is the fact that it's a CSS only solution a problem? Isn't that what CSS is for?

Reply:
The CSS solution is not the problem here. The problem lies that we add styles that only handle a specific use case.

I can't see the reason why not even check the above PR: https://github.com/WordPress/gutenberg/pull/40851. It's a suggestion that handles the use cases with any inner blocks included and also the `center` alignment.  

We shouldn't merge this before considering at least a universal solution. It might not be my PR, but this is what we should aim for - something that handles all..

Reply:
There have been three other attempts to solve this in a universal way: #35430 #40553, and #36681.

Reply:
What do we mean by universal way?

Reply:
> What do we mean by universal way?

I assume this means handling other cases - like maintaining the position of the blocks when you have a page numbers block and a next link but no previous link., or a previous link and a page numbers block, but no next link.

Reply:
Ah, I don't think we should handle the page numbers case because the design just doesn't work there and it's a lot more complex